corehq/apps/reports/views.py

```diff
@@ -990,9 +990,12 @@ def page_context(self):
 
     def post(self, request, *args, **kwargs):
         if self.scheduled_report_form.is_valid():
-            for k, v in self.scheduled_report_form.cleaned_data.items():
-                setattr(self.report_notification, k, v)
-
+            try:
+                self.report_notification.update_attributes(self.scheduled_report_form.cleaned_data.items())
+            except ValidationError as err:
+                kwargs['error'] = err.message
```

Comment:
Exception.message is removed in python 3

```suggestion
                kwargs['error'] = six.text_type(err)
```